enumerable.rb

```diff
@@ -0,0 +1,175 @@
+# rubocop:disable all
+module Enumerable
+  def my_each()
+    return to_enum(:my_each) unless block_given?
+
+    var = to_a
+    result = []
+    i = 0
+    while i < var.length
+      result << yield(var[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index()
+    return to_enum(:my_each_with_index) unless block_given?
+
+    var = to_a
+    i = 0
+    while i < var.length
+      result = yield(var[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    arr = []
+    my_each { |i| arr.push(i) if yield(i) }
+    arr
+  end
+
+  def my_all?(arg = nil)
+    var = to_a
+    result = true
+    var.my_each do |n|
+      if block_given?
+        if yield(n)
+          result = true
+        elsif !yield(n)
+          result = false
+        end
+      elsif arg.nil?
+        result = false if n.nil? || n == false
+      else
+        if arg === n
+          result = false
+          break
+        end
+      end
+    end
+    result
+  end
```

Comment:
There are a couple of problems with this method:

![image](https://user-images.githubusercontent.com/2739245/94324761-f294c200-ff60-11ea-9fa9-3af7c0b3bddd.png)

Your `#my_all` method should return false when the block returns false. You may use these definitions to test your code:

```
array = [0, 1, 2, 3, 4, 5]
false_block = proc { |num| num > 9 }
```

![image](https://user-images.githubusercontent.com/2739245/94324881-7f3f8000-ff61-11ea-8797-743eafc85d47.png)

You may use the same definiton of `array` as before to test this case.

![image](https://user-images.githubusercontent.com/2739245/94324907-9b432180-ff61-11ea-9aa4-e0812f826530.png)

This case is similar as before, but here you may use instead:

```
numbers = [1, 2i, 3.14]
```

![image](https://user-images.githubusercontent.com/2739245/94324941-ca599300-ff61-11ea-8319-e3b485d5ba93.png)

For this case, try with:

```
words = %w[dog door rod blade]
```

Finally:

![image](https://user-images.githubusercontent.com/2739245/94324972-f5dc7d80-ff61-11ea-9213-f92e6640d71a.png)

Here, you may use:

```
array = [3, 3, 3]
```